cpp/src/merge/merge.cu

```diff
@@ -0,0 +1,489 @@
+
+    auto left_end_zip_iterator = thrust::make_zip_iterator(thrust::make_tuple(left_side + left_size, left_indices + left_size));
+    auto right_end_zip_iterator = thrust::make_zip_iterator(thrust::make_tuple(right_side + right_size, right_indices + right_size));
+
+    rmm::device_vector<index_type> merged_indices(total_size);
+    
+    auto lhs_device_view = cudf::table_device_view::create(left_table, stream);
+    auto rhs_device_view = cudf::table_device_view::create(right_table, stream);
+
+    rmm::device_vector<cudf::order> d_column_order(column_order); 
+    
+    auto exec_pol = rmm::exec_policy(stream);
+    if (nullable){
+      rmm::device_vector<cudf::null_order> d_null_precedence(null_precedence);
+      
+      auto ineq_op =
+        cudf::experimental::detail::row_lexicographic_tagged_comparator<true>(*lhs_device_view,
+                                                                              *rhs_device_view,
+                                                                              d_column_order.data().get(),
+                                                                              d_null_precedence.data().get());
+      
+        thrust::merge(exec_pol->on(stream),
+                    left_begin_zip_iterator,
+                    left_end_zip_iterator,
+                    right_begin_zip_iterator,
+                    right_end_zip_iterator,
+                    merged_indices.begin(),
+                      [ineq_op] __device__ (index_type const & left_tuple,
+                                            index_type const & right_tuple) {
+                        return ineq_op(left_tuple, right_tuple);
+                    });			        
+    } else {
+      auto ineq_op =
+        cudf::experimental::detail::row_lexicographic_tagged_comparator<false>(*lhs_device_view,
+                                                                               *rhs_device_view,
+                                                                               d_column_order.data().get()); 
+        thrust::merge(exec_pol->on(stream),
+                    left_begin_zip_iterator,
+                    left_end_zip_iterator,
+                    right_begin_zip_iterator,
+                    right_end_zip_iterator,
+                    merged_indices.begin(),
+                      [ineq_op] __device__ (index_type const & left_tuple,
+                                            index_type const & right_tuple) {
+                        return ineq_op(left_tuple, right_tuple);
+                          
+                    });					        
+    }
+
+    CHECK_STREAM(stream);
+
+    return merged_indices;
+}
+
+} // namespace
+
+namespace cudf {
+namespace experimental { 
+namespace detail {
+
+//work-in-progress:
+//
+//generate merged column
+//given row order of merged tables
+//(ordered according to indices of key_cols)
+//and the 2 columns to merge
+//
+struct ColumnMerger
```

Comment:
CamelCase is reserved for test fixture names and template parameters in libcudf. Otherwise use lowercase_underscore naming.

```suggestion
struct column_merger
```

Reply:
Done.